python-avd/pyavd/_eos_designs/structured_config/network_services/ethernet_interfaces.py

```diff
@@ -58,6 +62,51 @@ def ethernet_interfaces(self: AvdStructuredConfigNetworkServicesProtocol) -> Non
         # Add interfaces used for Internet Exit policies
         self._set_internet_exit_policy_interfaces()
 
+    def _set_l3_port_channel_members(
+        self: AvdStructuredConfigNetworkServicesProtocol,
+        vrf: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem,
+    ) -> None:
+        """Set the structured_config for ethernet_interfaces which are members of l3_port_channels."""
+        for l3_port_channel in vrf.l3_port_channels:
+            # sub-interface for l3_port_channel cannot have member eth ports
+            # skip any logic to generate member port config for such sub-interfaces
+            if "." in l3_port_channel.name:
+                continue
+
+            channel_group_id = l3_port_channel.name.split("Port-Channel")[-1]
+            for member_intf in l3_port_channel.member_interfaces:
+                interface_description = member_intf.description
+                # derive values for peer from parent L3 port-channel
+                # if not defined explicitly for member interface
+                peer = member_intf.peer if member_intf.peer else l3_port_channel.peer
+                if not interface_description:
+                    interface_description = self.shared_utils.interface_descriptions.underlay_ethernet_interface(
+                        InterfaceDescriptionData(
+                            shared_utils=self.shared_utils,
+                            interface=member_intf.name,
+                            peer=peer,
+                            peer_interface=member_intf.peer_interface,
+                        ),
+                    )
+
+                ethernet_interface = EosCliConfigGen.EthernetInterfacesItem(
+                    name=member_intf.name,
+                    description=interface_description,
```

Comment:
An empty string could lead to failures on EOS, so we need to fall back to None.

```suggestion
                    description=interface_description or None,
```

Reply:
changed